[DCOS-57229] Add docs for Cassandra 2.7.0-3.11.4 release (#2540)
mesosphere/dcos-docs-site: pages/mesosphere/dcos/services/cassandra/2.7.0-3.11.4/release-notes/index.md

model: /mesosphere/dcos/services/cassandra/data.yml
render: mustache
---
# Release Notes for Cassandra Service version 2.7.0-3.11.4

## Updates

- Upgraded SDK library to version `0.57.0`. For more information see [SDK Release Notes](https://github.com/mesosphere/dcos-commons/releases/tag/0.57.0)
- Added support for [Secure JMX](/mesosphere/dcos/services/{{ model.serviceName }}/2.7.0-3.11.4/advanced/#secure-jmx-enterprise)

## Upgrading your cluster from 2.5.0-3.11.3 to 2.7.0-3.11.4

> - If upgrading from version `2.5.0-3.11.3`, you'll need to run the following command to successfully upgrade your Cassandra package: 

Reviewer: Let's add a note about how it's possible to skip an upgrade step
```suggestion
- It's possible to upgrade directly to `2.7.0-3.11.4` from version `2.5.0-3.11.3`. However, you'll need to run the following command to successfully upgrade your Cassandra package: 
```

Reviewer 2: Done

Author: thanks @SarahStegall

model: /mesosphere/dcos/services/cassandra/data.yml
render: mustache
---
# Release Notes for Cassandra Service version 2.7.0-3.11.4

## Updates

> - Upgraded SDK library to version `0.57.0`

Reviewer: Let's add a link to point to sdk release notes as it is significant
```suggestion
- Upgraded SDK library to version `0.57.0`. For more information see [SDK Release Notes](https://github.com/mesosphere/dcos-commons/releases/tag/0.57.0)
```

- Added support for [Secure JMX](/mesosphere/dcos/services/{{ model.serviceName }}/2.7.0-3.11.4/advanced/#secure-jmx-enterprise)

## Upgrading your cluster from 2.5.0-3.11.3 to 2.7.0-3.11.4

> - Run the following command to upgrade your Cassandra package: 

Reviewer: ```suggestion
- If upgrading from version `2.5.0-3.11.3`, you'll need to run the following command to successfully upgrade your Cassandra package: 
```

  ```
  dcos cassandra update start --package-version=2.7.0-3.11.4 --replace
  ```
  
<!--

> # Release Notes for Cassandra Service version 2.6.0-3.11.4

Reviewer: Actually, since this is the same minor version (and has a lot of content) let's keep the release notes for this version public.

## Improvements

- Service names are now validated with a regex.

# Release Notes for {{ model.techShortName }} Service version 2.4.0-3.0.16

## Bug Fixes

- Fix a bug where an out of date configuration ID would be selected when restarting or replacing pods. This could lead to configuration updates being reverted to the values with which the service was initially deployed. ([#2694](https://github.com/mesosphere/dcos-commons/pull/2694))

## Updates

> 

Reviewer: I would like to keep all these release notes as comments. Since we're deleting all the previous versions' directories and information, at least we can keep the release notes handy so that it's easy to find for developers. We often get asked by support if version X supports feature Y for a lot of our packages and going to release notes is a quick way to determine that. If we remove this, we're forced to go on a scavenger hunt in Github.

Reviewer 2: Cassie and I have been discussing this. Rather than delay the release of these documents while we decide the issue, I'm going to go with the release notes as is and ask @ck4adventure to merge them so we can get the docs out. We can always update the release notes later. I'll make a note to come back to this issue, because it will affect future docs releases as well. Thanks for bringing it to my attention; you raise a good point, but we need to formulate policy on this.
